pkg/crc/api/api_http.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func NewMux(config *crcConfig.Config, machine machine.Client, logger Logger, telemetry Telemetry) http.Handler {
-	handler := NewHandler(config, machine, logger, telemetry)
+	handler := NewHandler(config, machine, logger, telemetry, MachineClientUpdater)
```

Comment:
This commit seems to be about supporting multiple VMs running at the same time? As in the previous commits, I'd prefer to keep this kind of changes out of this PR, there's already a lot to get right without adding this kind of complication.

Reply:
> This commit seems to be about supporting multiple VMs running at the same time?

this is needed just for having different VM names, not necessarily enabling them to run at the same time.

since we pass the machine.Client to this function, it is tied to one _VM name_, but when the preset is changed, we need a new machine.Client with the new preset's VM name, earlier when all the VMs were named `crc` this was not needed